ui/apps/platform/src/services/CollectionsService.ts

```diff
@@ -188,12 +188,28 @@ export type CollectionDryRunResponse = {
  * @returns A list of deployments that are resolved by the collection.
  */
 export function dryRunCollection(
-    dryRunRequest: CollectionDryRunRequest
-    // TODO `ListDeployment` will make this impossible to paginate without loading the entire
-    // dataset client side. Ask [BE] if there is an efficient way to aggregate namespaces/clusters
-    // under a matching deployment name similar to the graphql query. An alternative might be to
-    // change the rendering of the list to not group deployments, but to sort alphabetically.
+    collectionRequest: CollectionRequest,
+    searchFilter: string,
+    page: number,
+    pageSize: number
 ): CancellableRequest<ListDeployment[]> {
+    const dryRunRequest: CollectionDryRunRequest = {
+        ...collectionRequest,
+        options: {
+            filterQuery: {
+                query: searchFilter,
+                pagination: {
+                    offset: page * pageSize,
+                    limit: pageSize,
+                    sortOption: {
```

Comment:
I think you can omit the `sortOption` in the API call which we should do instead if it works. I'd also make the `sortOption` an optional parameter to the function so that if we decide to reuse this elsewhere we have the ability to change it on demand.